src/IRCServer.py

```diff
@@ -304,7 +304,19 @@ def try_rejoin(self, event: EventManager.Event):
         if event["server_id"] == self.id and event["channel_name"
                 ] in self.attempted_join:
             self.send_join(event["channel_name"], event["key"])
-    def send_join(self, channel_name: str, key: str=None) -> IRCLine.Line:
+    def send_join(self, channel_name: str, key: str="") -> IRCLine.Line:
+        if "," in channel_name:
+            # Multiple channels
+            chans = channel_name.split(",")
+            channel_name = list()
+            for chan in chans:
+                if not self.channels.contains(chan):
+                    channel_name.append(chan)
+            channel_name = ",".join(channel_name)
+        else:
+            if self.channels.contains(channel_name):
+                return self.send("")
```

Comment:
Invalid. We really should return `None` here but I'm more inclined to say that we should leave the "are we already in this channel" to specific modules - this function should just be an alias for `JOIN #channel pass`

Reply:
Alright, I can reverse this, it's not exactly like it's a bandwidth hog sending a bunch of JOIN requests! :)

Reply:
Resolved, and I fixed a logic error.